app/src/main/java/com/sun/basic_japanese/alphabet/eachpage/AlphabetPagePresenter.kt

```diff
@@ -0,0 +1,59 @@
+package com.sun.basic_japanese.alphabet.eachpage
+
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.WORD_DEFAULT_GROUP
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.WORD_MAX_GROUP
+import com.sun.basic_japanese.data.model.Alphabet
+import com.sun.basic_japanese.data.model.AlphabetsResponse
+import com.sun.basic_japanese.data.repository.AlphabetRepository
+import com.sun.basic_japanese.data.source.OnDataLoadedCallback
+
+class AlphabetPagePresenter(
+    private val pageView: AlphabetPageContract.View,
+    private val alphabetRepository: AlphabetRepository
+) : AlphabetPageContract.Presenter {
+
+    private var alphabetRawItems = mutableListOf<Alphabet>()
+
+    override fun prepareAlphabetData() {
+
+        alphabetRepository.getAllAlphabets(object : OnDataLoadedCallback<AlphabetsResponse> {
+            override fun onSuccess(data: AlphabetsResponse) {
+                alphabetRawItems = data.alphabets
+                prepareData()
```

Comment:
tôi nghĩ đoạn này nên viết theo kiểu giá trị trả về thì sẽ clear hơn. 
```suggestion
                val alphaRawItems = getAlphaRawData(data.alphabets)
                pageView.show(alphaRawItems)
```